### PR TITLE
flamenco, fuzz: VM interpreter fuzzer

### PR DESCRIPTION
**Pull request by ravyu-jump**

### CODE REVIEW

File: src/flamenco/runtime/tests/fd_vm_test.c [hunk 1]
```diff
@@ -95,6 +118,199 @@ fd_exec_vm_validate_test_run( fd_exec_instr_test_runner_t * runner,
...
+  if ( fd_vm_validate( vm ) != FD_VM_SUCCESS ) {
+    effects->error = -1;
+    break;
+  }
+
+  // Override some execution state values from the interp fuzzer input
+  // This is so we can test if the interp mutates any of these erroneously
+  // Those that are not overridden are because they are set in fd_vm_setup_state_for_execution
+  vm->reg[0]  = input->vm_ctx.r0;
+  // vm->reg[1]  = input->vm_ctx.r1;
```
Comment (topointon-jump) [inline src/flamenco/runtime/tests/fd_vm_test.c:227]: let's try and not hard-code this

File: src/flamenco/runtime/tests/fd_vm_test.c [hunk 2]
```diff
@@ -95,6 +118,199 @@ fd_exec_vm_validate_test_run( fd_exec_instr_test_runner_t * runner,
...
+  vm->reg[2]  = input->vm_ctx.r2;
+  vm->reg[3]  = input->vm_ctx.r3;
+  vm->reg[4]  = input->vm_ctx.r4;
+  vm->reg[5]  = input->vm_ctx.r5;
+  vm->reg[6]  = input->vm_ctx.r6;
+  vm->reg[7]  = input->vm_ctx.r7;
+  vm->reg[8]  = input->vm_ctx.r8;
+  vm->reg[9]  = input->vm_ctx.r9;
+  // vm->reg[10] = input->vm_ctx.r10;
```
Comment (topointon-jump) [inline src/flamenco/runtime/tests/fd_vm_test.c:236]: let's try and not hard-code this

File: src/flamenco/runtime/tests/fd_vm_test.c [hunk 3]
```diff
@@ -95,6 +118,199 @@ fd_exec_vm_validate_test_run( fd_exec_instr_test_runner_t * runner,
...
+  for( ulong i=0; i< fd_sbpf_syscalls_slot_cnt(); i++ ){
+    fd_sbpf_syscalls_t * syscall = fd_sbpf_syscalls_query( syscalls, syscalls[i].key, NULL ); // TODO: can just use syscalls[i].func directly?
+    if ( !syscall ) {
+      continue;
+    }
+    syscall->func = fd_vm_syscall_noop;
+  }
+
+  /* Setup trace */
+  const uint ENABLE_TRACING = 0; // Set to 1 to enable tracing
```
[the remark this answers is not in the document]
Comment (topointon-jump) [reply src/flamenco/runtime/tests/fd_vm_test.c:180]: add a tracing_enabled field in vm context protobuf

**Review by topointon-jump — DISMISSED** [review]

File: src/flamenco/runtime/tests/fd_vm_test.c [hunk 4]
```diff
@@ -4,6 +4,29 @@
...
+                    ulong arg0,
+                    ulong arg1,
+                    ulong arg2,
+                    ulong arg3,
+                    ulong arg4,
+                    ulong* _ret){
+  /* TODO: have input message determine CUs to deduct? 
+  fd_vm_t * vm = (fd_vm_t *) _vm;
+  vm->cu = vm->cu - 5;
+  */
```
[the remark this answers is not in the document]
Comment (ravyu-jump) [reply src/flamenco/runtime/tests/fd_vm_test.c:18]: Will work on a separate PR for this along with feature flags

**Comment by ravyu-jump** [comment]
@topointon-jump here are the changes since review:

https://github.com/firedancer-io/firedancer/compare/a9248907237345235ca9a2eddcdfabece4e892b5..8da7f6a343150a495d6f038ff19d6d9aa4c29cb7

**Review by topointon-jump — APPROVED** [review]

**Pull request merged**